src/chocolatey.benchmark/helpers/ManagedProcessHelper.cs

```diff
@@ -0,0 +1,233 @@
+
+                var foundProcess = FindPidFromIndexedProcessName(processName);
+                
+                if (foundProcess == null)
+                {
+                    break;
+                }
+
+                nextProcess = foundProcess;
+                tree.Processes.AddLast(nextProcess.ProcessName);
+            }
+
+            return tree;
+        }
+
+        public static string GetParent(Process process = null)
+        {
+            if (process == null)
+            {
+                process = Process.GetCurrentProcess();
+            }
+
+            Process nextProcess = null;
+
+            while (true)
+            {
+                var processId = nextProcess?.Id ?? process.Id;
+
+                var processName = FindIndexedProcessName(processId);
+
+                if (string.IsNullOrEmpty(processName))
+                {
+                    break;
+                }
+
+                var foundProcess = FindPidFromIndexedProcessName(processName);
+
+                if (foundProcess == null)
+                {
+                    break;
+                }
+
+                nextProcess = foundProcess;
+            }
+
+            return nextProcess?.ProcessName;
+
+            //var processName = FindIndexedProcessName(process.Id);
+
+            //if (string.IsNullOrEmpty(processName))
+            //{
+            //    return null;
+            //}
+
+            //var parentProcess = FindPidFromIndexedProcessName(processName);
+
+            //if (parentProcess == null)
+            //{
+            //    return null;
+            //}
+
+            //var topLevelProcess = GetParent(parentProcess);
+
+            //if (topLevelProcess == null)
+            //{
+            //    return parentProcess.ProcessName;
+            //}
+
+            //return topLevelProcess;
+        }
+
+        public static string GetParentFiltered(Process process = null)
+        {
+            if (process == null)
+            {
+                process = Process.GetCurrentProcess();
+            }
+
+            Process nextProcess = null;
+            Process selectedProcess = null;
+
+            while (true)
+            {
+                var processId = nextProcess?.Id ?? process.Id;
+
+                var processName = FindIndexedProcessName(processId);
+
+                if (string.IsNullOrEmpty(processName))
+                {
+                    break;
+                }
+
+                var foundProcess = FindPidFromIndexedProcessName(processName);
+
+                if (foundProcess == null)
+                {
+                    break;
+                }
+
+                nextProcess = foundProcess;
+
+                if (!IsIgnoredParent(nextProcess.ProcessName))
+                {
+                    selectedProcess = nextProcess;
+                }
+            }
+
+            return selectedProcess?.ProcessName;
+
+            //var processName = FindIndexedProcessName(process.Id);

```

Comment:
These commented out code is similar to the above comment, and can be safely removed.